src/codeu/chat/client/ClientUser.java

```diff
@@ -84,6 +89,19 @@ public boolean signOutUser() {
     return hadCurrent;
   }
 
+  public void setCurrent(User user) {
+    current = user;
+  }
+
+  public boolean checkPassword(String password) {
+    String test = password + getCurrent().getSalt();
```

Comment:
so this is why i don't think you should verify this on the client side - wouldn't getCurrent() be null when you first open the chat client? How do you know which user is going to log in without figuring out the username?